lib/algae/result.ex

```diff
@@ -0,0 +1,323 @@
+defmodule Algae.Result do
+  @moduledoc ~S"""
+  Data type representing a result.
+  A result can either be a successful `Ok` or an unsuccessful `Err`.
```

Comment:
@icidasset if you could expand this section on how it's different from `Algae.Either`, that would be helpful, too 😄